src/logging.cpp

```diff
@@ -160,6 +160,7 @@ const CLogCategoryDesc LogCategories[] =
     {BCLog::I2P, "i2p"},
     {BCLog::IPC, "ipc"},
     {BCLog::LOCK, "lock"},
+    {BCLog::BLOCK, "block"},
```

Comment:
Is there a guideline on which log category to use within validation? There is `BENCH` used for benchmarking validation. There is `VALIDATION` used for storage and validation. Now there is `BLOCK` used for `CheckBlockIndex`.

Reply:
Not that I know of. I published a doc at https://github.com/jonatack/bitcoin-development/blob/master/logging.md#log-categories that is an updated version of this [Bitcoin Stack Exchange answer](https://bitcoin.stackexchange.com/questions/66892/what-are-the-debug-categories/66895#66895) in 2017 by @achow101 and am considering adding it to the RPC logging help.

Reply:
Is there a reason to not re-use the BENCH or VALIDATION category? Seems overkill to introduce a new category for a single log line that is mostly only enabled for devs/debugging.

Reply:
We did that for instance with MEMPOOLREJ (1 line), QT (1 line), LOCK (1 line), PROXY (1 line), RAND (2 lines), and COINDB (3 lines).

My thought was that VALIDATION is too busy and BENCH would be ok but seems not directly related to consistency checks.  We could also rename CMPCTBLOCK to BLOCK and add it to that group.

Reply:
I think a BLOCK category makes sense in the sense that "block storage" is separate from "validation". We have a separate label for it too. If it is planned to be used for other messages too.

On the other hand this particular functionality is already enabled with a special option (`-checkblockindex`), so for this specifically I'm not sure also adding a separate category makes sense.

> RAND (2 lines)

I think this one should go into a more general UTIL category at some point.

> QT (1 line)

One line that captures everything from Qt's internal logging system :smile:

Reply:
Emoji vote?
- :rocket: for BENCH
- :heart: for BLOCK
- :tada: for BLOCK and move CMPCTBLOCK into it too

Reply:
> I think a BLOCK category makes sense in the sense that "block storage" is separate from "validation". We have a separate label for it too. If it is planned to be used for other messages too.
> 
> On the other hand this particular functionality is already enabled with a special option (`-checkblockindex`), so for this specifically I'm not sure also adding a separate category makes sense.

Right. It doesn't seem ideal to use `BCLog::VALIDATION` or `BCLog::BENCH`, and `LogPrintf` would require bitcoind to be restarted to turn off the `-checkblockindex` config option. Maybe rename the low-frequency `BCLog::CMPCTBLOCK` category to `BCLog::BLOCK` and use that, but it would be a bit more disruptive than the current proposal. Or maybe I should close this and just use the change locally.

Reply:
> Looks like #23235 did some of the clean-ups here?

Looked at it, but that change looks like it needs cleaning up and given how this simple change couldn't go through, I didn't attempt it.